### PR TITLE
feat(notifier): Add Jira notifier configuration

### PR DESCRIPTION
**Pull request by wkl3nk**
Providing a way to configure a Jira notifier to interact with a Jira server at the end of an ORT run, e.g. to create a comment to an existing Jira issue about the result of the ORT run.

### CODE REVIEW

File: workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt [hunk 1]
```diff
@@ -70,17 +72,36 @@ class NotifierRunner {
...
+                        username = jiraRestClientUsername,
+                        password = jiraRestClientPassword
+                    )?.mapToOrt()
+            } else {
+                JiraConfiguration()
             }
 
+            val notifierConfiguration = NotifierConfiguration(
+                mail = sendMailConfiguration,
+                jira = jiraConfiguration
```
Comment (MarcelBochtler) [inline workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt:102]: IMO we shouldn't use the default mail and Jira configurations, if the user didn't configure them.
I think both values should be `null` if they are not configured. This causes a different behavior in the ORT notifier, where `null` is handled separately:
https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/main/kotlin/Notifier.kt#L46-L47

File: README.md [hunk 2]
```diff
@@ -98,13 +98,13 @@ without TLS.**
...
-|----------------|----------------------------------|-------------------|
-| ORT Server API | http://localhost:8080/swagger-ui |                   |
-| Keycloak       | http://localhost:8081            | admin:admin       |
-| PostgreSQL     | http://localhost:5433            | postgres:postgres |
-| RabbitMQ       | http://localhost:15672           | admin:admin       |
-| Graphite       | http://localhost:8888            | root:root         |
+| Service        | URL                              | Credentials                                             |
+|----------------|----------------------------------|---------------------------------------------------------|
+| ORT Server API | http://localhost:8080/swagger-ui |                                                         |
+| Keycloak       | http://localhost:8081            | Administrator: admin:admin<br/>User: ort-admin:password |
```
Comment (MarcelBochtler) [inline README.md:104]: This change should be in a separate commit.

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]

File: workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt (hunk 1 above)
Comment (wkl3nk) [reply workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt:102]: Actually, we have no other choice than providing default Mail and Jira configurations, if the user did not explicitly configure one, because, as you already pointed out, the _Notifier_ from the oss-review-toolkit "injects" variables _mailClient_ and _jiraClient_ into the script _reporter-notification.kts_ (ort-server-config). If only **one** of these variables is not injected, then the complete kotlin script **will not compile**, and neither Mail nor Jira notifications are created at all.

_Unresolved reference: mailClient
Unresolved reference: jiraClient_

This raises the question, if there should be a refactoring in a way that there is one kotlin reporter script for Mail notifications, and another one for Jira notifications. Then, depending on which configurations are present, one or the other script could be executed or not. But IMO this would be a separate explicit change request.

Comment (sschuberth) [reply workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt:102]: > If only one of these variables is not injected, then the complete kotlin script will not compile

Actually, I believe these tests prove the contrary:

[Run a script without a Jira configuration](https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/test/kotlin/NotifierTest.kt#L41-L49)

[Run a script without a mail configuration](https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/test/kotlin/NotifierTest.kt#L61-L69)

File: README.md (hunk 2 above)
Comment (wkl3nk) [reply README.md:104]: Done.

File: workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt (hunk 1 above)
Comment (wkl3nk) [reply workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt:100]: You convinced me. Let's stick to the contract. If a user provides a notifier script that contains notification code for both mail and jira, but forgot to take care for proper of mail and jira, then better let the whole script crash.

**Review by bs-ondem — APPROVED** [review]

File: workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt [hunk 3]
```diff
@@ -70,17 +70,36 @@ class NotifierRunner {
...
+            }
+
+            val jiraRestClientUsername = config.jira?.jiraRestClientConfiguration?.username?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+            val jiraRestClientPassword = config.jira?.jiraRestClientConfiguration?.password?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+
+            val jiraConfiguration = if (jiraRestClientUsername != null && jiraRestClientPassword != null) {
```
Comment (oheger-bosch) [inline workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt:89]: Nit: You could drop the `else { null }` part by replacing the `if` construct with `takeIf`, e.g. something like

```
val jiraConfiguration = config.jira?.jiraRestClientConfiguration?.takeIf { 
  jiraRestClientUsername != null && jiraRestClientPassword != null
}?.copy(username = jiraRestClientUsername, password = jiraRestClientPassword)?.mapToOrt()
```

File: workers/notifier/src/test/kotlin/NotifierRunnerTest.kt [hunk 4]
```diff
@@ -62,6 +64,13 @@ class NotifierRunnerTest : WordSpec({
                     useSsl = false,
                     fromAddress = "no-reply@oss-review-toolkit.org"
                 )
+            ),
```
Comment (oheger-bosch) [inline workers/notifier/src/test/kotlin/NotifierRunnerTest.kt:67]: Unfortunately, the test is not really meaningful - but this is out of scope of this PR.

**Comment by mnonnenmacher** [comment]
@wkl3nk No need to recreate the PR, we can just dismiss Marcel's review in such cases.
